pkg/resource/key/hooks.go

```diff
@@ -42,3 +50,51 @@ func GetDeletePendingWindowInDays(
 
 	return int64(pendingWindowInt)
 }
+
+// customUpdate is the implementation of update operation for KMS Key resource.
+// This operation will return a Terminal error if the update is not for 'Policy'
+// , 'Tags' or 'BypassPolicyLockoutSafetyCheck' because KMS Key only supports
+// updating those three fields.
+//
+func (rm *resourceManager) customUpdate(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (updated *resource, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.customUpdate")
+	defer func() {
+		exit(err)
+	}()
+	updatedRes := rm.concreteResource(desired.DeepCopy())
+	updatedRes.SetStatus(latest)
+
+	if delta.DifferentExcept("Spec.Policy", "Spec.Tags", "Spec.BypassPolicyLockoutSafetyCheck") {
```

Comment:
I agree that would be a consistent way of handling this. But when i look at the current implementation of `handleImmutableFieldsChangedCondition` it does not seem to work.
a) It checks `delta.DifferentAt("FieldName")` instead of correct way `delta.DifferentAt("Spec.FieldName")`
b) I also believe that the condition set should be Terminal instead of Advisory
c) The update call should error out incase immutable field is modified.

All the 3 mentioned changes are missing currently. Once we make a decision on above 3 points, i will include those changes in this PR. Holding until then.

Reply:
https://github.com/aws-controllers-k8s/code-generator/pull/350